closure.ts

```diff
@@ -0,0 +1,348 @@
+import { Stmt, Expr, Type, UniOp, BinOp, Literal, Program, FunDef, VarInit, Class, SourceLocation } from './ast';
+import { TypeCheckError } from './error_reporting'
+import { NUM, BOOL, NONE, CLASS } from './utils';
+
+let closureClassCounter = 0
+
+export function translateClosuresToClasses(program : Program<SourceLocation>): Program<SourceLocation> {
+  closureClassCounter = 0
+
+  program.funs.forEach(func => translateClosuresInFunc(func, program))
+  // TODO methods in class might also contain closures
+
+  return program
+}
+
+function translateClosuresInFunc(f: FunDef<SourceLocation>, program: Program<SourceLocation>) {
+  const body: Stmt<SourceLocation>[] = []
+  for (const stmt of f.body) {
+    if (stmt.tag == "closure") {
+      // global variable
+      // def f
+      //     def g <-- closure
+      const g = stmt.func
+
+      // TODO
```

Comment:
Again, not sure if the TODO has been done or yet to be done.

Reply:
Yes. We forgot to delete this TODO.